### PR TITLE
Include ASP packages in offline cache

### LINKED ISSUE #4957: Include ASP.NET in Offline Package Cache
Currently we're able to restore a Console application offline in CLI. We need to also make sure that the default Web template is also restorable offline.

This will impact the offline archiving build step. We need to make sure this continues to work well.

### PR DESCRIPTION
**Pull request by jgoshi**
Fixes #4957 
@piotrpMSFT @livarcocc @krwq @jonsequitur

### CODE REVIEW

**Comment by jgoshi** [comment]
test Windows_NT x64 Release Build

**Comment by jgoshi** [comment]
@piotrpMSFT @krwq Can one of you take a look if you have a chance?

File: build/Microsoft.DotNet.Cli.Prepare.targets [hunk 1]
```diff
@@ -128,6 +132,7 @@
     <PropertyGroup>
       <CoreSetupBlobRootUrl>https://dotnetcli.azureedge.net/dotnet/$(CoreSetupChannel)</CoreSetupBlobRootUrl>
       <SharedFrameworkArchiveBlobRootUrl>$(CoreSetupBlobRootUrl)/Binaries/$(SharedFrameworkVersion)</SharedFrameworkArchiveBlobRootUrl>
+      <NuGetPackagesArchiveBlobRootUrl>$(CoreSetupBlobRootUrl)/Binaries/$(NuGetPackagesArchiveVersion)</NuGetPackagesArchiveBlobRootUrl>
```
Comment (livarcocc) [inline build/Microsoft.DotNet.Cli.Prepare.targets]: Does this mean that we are going to have potentially a bunch of weird folders in the blob? I wonder if we should just keep adding the lzma archives in the shared framework version to which they belong. Just change the lzma filename itself with the hash.

What do you think?

File: test/dotnet.Tests/GivenThatTheUserIsRunningDotNetForTheFirstTime.cs [hunk 2]
```diff
@@ -78,19 +78,75 @@ public void ItShowsTheAppropriateMessageToTheUser()
...
 
         [Fact]
-        public void ItCreatesASentinelFileUnderTheNuGetCacheFolder()
+        public void ItRestoresTheNuGetPackagesToTheNuGetCacheFolder()
         {
             _nugetCacheFolder
                 .Should()
                 .HaveDirectory("microsoft.netcore.app");
+
+            _nugetCacheFolder
```
Comment (livarcocc) [inline test/dotnet.Tests/GivenThatTheUserIsRunningDotNetForTheFirstTime.cs]: I wonder if we should make this a list and add a Should().HaveDirectories() to the DirectoryInfoAssertions, if there isn't one already.

File: build_projects/dotnet-cli-build/GenerateNuGetPackagesArchiveVersion.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                        dataToHash += $"{metadata.Name}={metadata.Value};";
+                    }
+                }
+            }
+
+            Log.LogMessage($"NuGet Packages Archive Data To Hash: '{dataToHash}'");
+
+            var sha256 = SHA256.Create();
+            var hashBytes = sha256.ComputeHash(Encoding.Unicode.GetBytes(dataToHash));
+            Version = GetHashString(hashBytes);
```
Comment (livarcocc) [inline build_projects/dotnet-cli-build/GenerateNuGetPackagesArchiveVersion.cs]: What will be the max length of this Version? Just to make sure we won't end up running up out of the 256 for the path to this file.

File: build/Microsoft.DotNet.Cli.Prepare.targets (hunk 1 above)
Comment (jgoshi) [inline build/Microsoft.DotNet.Cli.Prepare.targets]: Ok

File: test/dotnet.Tests/GivenThatTheUserIsRunningDotNetForTheFirstTime.cs (hunk 2 above)
Comment (jgoshi) [inline test/dotnet.Tests/GivenThatTheUserIsRunningDotNetForTheFirstTime.cs]: Will do

File: build_projects/dotnet-cli-build/GenerateNuGetPackagesArchiveVersion.cs (hunk 3 above)
Comment (jgoshi) [inline build_projects/dotnet-cli-build/GenerateNuGetPackagesArchiveVersion.cs]: 256 bits right? I don't know what else I can use to ensure a shorter unique hash?

**Comment by jgoshi** [comment]
@srivatsn

**Pull request merged**